geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HScanDunitTest.java

```diff
@@ -0,0 +1,299 @@
+  @Before
+  public void testSetup() {
+    addIgnoredException(FunctionException.class);
+    String[] redisPortsAsStrings = new String[redisPorts.length];
+
+    for (int i = 0; i < redisPorts.length; i++) {
+      redisPortsAsStrings[i] = String.valueOf(redisPorts[i]);
+    }
+
+    DUnitSocketAddressResolver dnsResolver =
+        new DUnitSocketAddressResolver(redisPortsAsStrings);
+
+    ClientResources resources = ClientResources.builder()
+        .socketAddressResolver(dnsResolver)
+        .build();
+
+    redisClient = RedisClient.create(resources, "redis://localhost");
+    redisClient.setOptions(ClientOptions.builder()
+        .autoReconnect(true)
+        .build());
+
+    connection = redisClient.connect();
+    commands = connection.sync();
+    commands.hset(HASH_KEY, INITIAL_DATA_SET);
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    commands.quit();
+
+    server1.stop();
+    server2.stop();
+    server3.stop();
+  }
+
+  @Test
+  public void should_allow_hscan_iteration_to_complete_successfully_given_server_crashes_during_iteration()
+      throws ExecutionException, InterruptedException {
+
+    AtomicBoolean keepCrashingVMs = new AtomicBoolean(true);
+    AtomicInteger numberOfTimesServersCrashed = new AtomicInteger(0);
+
+    Future<Void> hScanFuture =
+        executor.runAsync(
+            () -> doHScanContinuallyAndAssertOnResults(keepCrashingVMs,
+                numberOfTimesServersCrashed));
+
+    Future<Void> crashingVmFuture =
+        executor.runAsync(
+            () -> crashAlternatingServers(keepCrashingVMs, numberOfTimesServersCrashed));
+
+
+    hScanFuture.get();
+    crashingVmFuture.get();
+  }
+
+
+  private static void doHScanContinuallyAndAssertOnResults(AtomicBoolean keepCrashingServers,
+      AtomicInteger numberOfTimesServersCrashed) {
+    int numberOfAssertionsCompleted = 0;
+
+    ScanCursor scanCursor = new ScanCursor("0", false);
+    List<String> allEntries = new ArrayList<>();
+    MapScanCursor<String, String> result;
+
+    while (numberOfAssertionsCompleted < 3 || numberOfTimesServersCrashed.get() < 3) {
+
+      allEntries.clear();
+      scanCursor.setCursor("0");
+      scanCursor.setFinished(false);
+
+      try {
+        do {
+          result = commands.hscan(HASH_KEY, scanCursor);
+          scanCursor.setCursor(result.getCursor());
+          Map<String, String> resultEntries = result.getMap();
+
+          resultEntries
+              .entrySet()
+              .forEach(
+                  entry -> allEntries.add(entry.getKey()));
+
+        } while (!result.isFinished());
+
+        assertThat(allEntries).containsAll(INITIAL_DATA_SET.keySet());
+        numberOfAssertionsCompleted++;
+
+      } catch (RedisCommandExecutionException ignore) {
+      } catch (RedisException ex) {
+        if (ex.getMessage().contains("Connection reset by peer")) {// ignore error
+        } else {
+          throw ex;
+        }
+      }
+    }
+    keepCrashingServers.set(false);
```

Comment:
I think if you look closer... this only gets set to false  when control leaves the while with the while (numberOfAssertionsCompleted < 3 || numberOfTimesServersCrashed.get() < 3)  logic.  so it will not be set to false (and stop the vms from crashing). until there have been at least 3 assertions and at least 3 vm crashes.